lib/ansible/modules/database/mongodb/mongodb_replicaset.py

```diff
@@ -0,0 +1,407 @@
+    type: bool
+    default: no
+  ssl_cert_reqs:
+    description:
+    - Specifies whether a certificate is required from the other side of the connection, and whether it will be validated if provided.
+    type: str
+    default: CERT_REQUIRED
+    choices: [ CERT_NONE, CERT_OPTIONAL, CERT_REQUIRED ]
+  arbiter_at_index:
+    description:
+    - Identifies the position of the member in the array that is an arbiter.
+    type: str
+  chainingAllowed:
+    description: >
+    - When I(settings.chainingAllowed=true), the replicaset allows secondary members to replicate from other
+      secondary members.
+    - When I(settings.chainingAllowed=false), secondaries can replicate only from the primary.
+    type: bool
+    default: yes
+  heartbeatTimeoutSecs:
+    description:
+    - Number of seconds that the replicaset members wait for a successful heartbeat from each other.
+    - If a member does not respond in time, other members mark the delinquent member as inaccessible.
+    - The setting only applies when using I(protocolVersion=0). When using I(protocolVersion=1) the relevant
+      setting is I(settings.electionTimeoutMillis).
+    type: int
+    default: 10
+  electionTimeoutMillis:
+    description:
+    - The time limit in milliseconds for detecting when a replicaset's primary is unreachable.
+    type: int
+    default: 10000
+  protocolVersion:
+    description: Version of the replicaset election protocol.
+    type: int
+    choices: [ 0, 1 ]
+    default: 1
+notes:
+- Requires the pymongo Python package on the remote host, version 2.4.2+. This
+  can be installed using pip or the OS package manager. @see U(http://api.mongodb.org/python/current/installation.html)
+requirements:
+- pymongo
+'''
+
+EXAMPLES = r'''
+# Create a replicaset called 'rs0' with the 3 provided members
+- name: Ensure replicaset rs0 exists
+  mongodb_replicaset:
+    login_host: localhost
+    login_user: admin
+    login_password: admin
+    replica_set: rs0
+    members:
+    - mongodb1:27017
+    - mongodb2:27017
+    - mongodb3:27017
+  when: groups.mongod.index(inventory_hostname) == 0
+
+# Create two single-node replicasets on the localhost for testing
+- name: Ensure replicaset rs0 exists
+  mongodb_replicaset:
+    login_host: localhost
+    login_port: 3001
+    login_user: admin
+    login_password: secret
+    login_database: admin
+    replica_set: rs0
+    members: localhost:3001
+    validate: no
+
+- name: Ensure replicaset rs1 exists
+  mongodb_replicaset:
+    login_host: localhost
+    login_port: 3002
+    login_user: admin
+    login_password: secret
+    login_database: admin
+    replica_set: rs1
+    members: localhost:3002
+    validate: no
+'''
+
+RETURN = r'''
+mongodb_replicaset:
+  description: The name of the replicaset that has been created.
+  returned: success
+  type: str
+'''
+
+from copy import deepcopy
+
+import os
+import ssl as ssl_lib
+import traceback
```

Comment:
This is no longer needed. Ansible adds traceback information automagically.
```suggestion
```

Reply:
Removed.